extension/observer/ecsobserver/ecsmock/service.go

```diff
@@ -0,0 +1,195 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecsmock
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/ecs"
+)
+
+// PageLimit defines number of items in a single page for different APIs.
+// Those numbers can be found on the Input and and Output struct comments.
+// Call DefaultPageLimit() to config the mock to use numbers same as the actual AWS API.
+type PageLimit struct {
+	ListTaskOutput                 int // default 100, max 100
+	DescribeTaskInput              int // max 100
+	ListServiceOutput              int // default 10, max 100
+	DescribeServiceInput           int // max 10
+	DescribeContainerInstanceInput int // max 100
+}
+
+func DefaultPageLimit() PageLimit {
+	return PageLimit{
+		ListTaskOutput:                 100,
+		DescribeTaskInput:              100,
+		ListServiceOutput:              10,
+		DescribeServiceInput:           10,
+		DescribeContainerInstanceInput: 100,
+	}
+}
+
+// Cluster implements both ECS and EC2 API for a single cluster.
+type Cluster struct {
+	taskList []*ecs.Task
+	taskMap  map[string]*ecs.Task
+	limit    PageLimit
+}
+
+// NewCluster creates a mock ECS cluster with default limits.
+func NewCluster() *Cluster {
+	return &Cluster{
+		taskMap: make(map[string]*ecs.Task),
+		limit:   DefaultPageLimit(),
+	}
+}
+
+// API Start
```

Comment:
Pls help to enhance this comment.

Reply:
This is a marker for making start and end of some block so it's easier for people to navigate the code. And if you have extra config in editor, it can helps folding the code.